lib/spack/spack/util/atomic_update.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import ctypes
+import os
+from contextlib import contextmanager
+from typing import Optional
+
+import llnl.util.filesystem as fs
+from llnl.util.symlink import symlink
+
+libc: Optional[ctypes.CDLL] = None
+try:
+    libc = ctypes.CDLL("/lib64/libc.so.6", 0x04)  # 0x04 is RTLD_NOLOAD
```

Comment:
Uh, don't open by path. Just do CDLL(None) and you get a handle for the Python process which links against libc.

Reply:
Did not realize I could use `None`, I was switching it to `"libc.so.6"` instead of the full path but I guess this is cleaner for some far-off future that libc changes.

I think this will also fix the code coverage issue, since the reason I was switching to not use a path was that libc lives in a different location on ubuntu and that's where our CI runs.